includes/Classifai/Services/LanguageProcessing.php

```diff
@@ -13,4 +15,91 @@ class LanguageProcessing extends Service {
 	public function __construct() {
 		parent::__construct( __( 'Language Processing', 'classifai' ), 'language_processing', [ 'Classifai\Providers\Watson\NLU' ] );
 	}
+
+	/**
+	 * Init service for Language Processing.
+	 */
+	public function init() {
+		parent::init();
+		add_action( 'rest_api_init', [ $this, 'register_endpoints' ] );
+	}
+
+	/**
+	 * Create endpoints for Language Processing.
+	 *
+	 * @since 1.8.0
+	 */
+	public function register_endpoints() {
+		register_rest_route(
+			'classifai/v1',
+			'generate-tags/(?P<id>\d+)',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'generate_post_tags' ],
+				'args'                => array(
+					'id' => array(
+						'required'          => true,
+						'type'              => 'integer',
+						'sanitize_callback' => 'absint',
+						'description'       => esc_html__( 'Post ID to generate tags.', 'classifai' ),
+					),
+				),
+				'permission_callback' => function () {
+					return current_user_can( 'edit_posts' );
+				},
```

Comment:
This is a tricky one. Right now, if Language Processing is turned on, we automatically classify content on save, without ever checking what permissions a user has, I think because the user is not actually assigning terms, the AI/bot is doing that.

We also are currently using [`current_user_can( 'edit_posts' )`](https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/Watson/NLU.php#L213) to determine if we should show that Classifai panel in Gutenberg, which I'm assuming is why were doing the same here.

This might be worth a broader discussion on permissions within Classifai but feels like if we want to keep the same behavior, we don't necessarily care about a user's permission to assign terms, just if they have permission to edit the piece of content. So I think `edit_post` should suffice here.

Reply:
I updated code to use `current_user_can( 'edit_post', $post_id )`, same at display Classifai panel as well.